shared/services/rocketpool/client.go

```diff
@@ -300,32 +301,23 @@ func (c *Client) UpdatePrometheusConfiguration(settings map[string]string) error
 		return fmt.Errorf("Error expanding Prometheus config file path: %w", err)
 	}
 
-	// Set the environment variables defined in the user settings for metrics
-	oldValues := map[string]string{}
-	for varName, varValue := range settings {
-		oldValues[varName] = os.Getenv(varName)
-		os.Setenv(varName, varValue)
-	}
-
-	// Read and substitute the template
-	contents, err := envsubst.ReadFile(prometheusTemplatePath)
+	// Parse the template
+	t, err := template.ParseFiles(prometheusTemplatePath)
 	if err != nil {
-		return fmt.Errorf("Error reading and substituting Prometheus configuration template: %w", err)
-	}
-
-	// Unset the env vars
-	for name, value := range oldValues {
-		os.Setenv(name, value)
+		return fmt.Errorf("Error reading Prometheus configuration template: %w", err)
 	}
 
-	// Write the actual Prometheus config file
-	err = os.WriteFile(prometheusConfigPath, contents, 0664)
+	// Create the output file
+	runtimeFile, err := os.OpenFile(prometheusConfigPath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0664)
 	if err != nil {
-		return fmt.Errorf("Could not write Prometheus config file to %s: %w", shellescape.Quote(prometheusConfigPath), err)
+		return fmt.Errorf("Could not open Prometheus config file %s for writing: %w", shellescape.Quote(prometheusConfigPath), err)
 	}
-	err = os.Chmod(prometheusConfigPath, 0664)
```

Comment:
I vaguely remember having to add this to work around `umask`, does the new stuff obviate that?

Reply:
I feel like the only way you'd end up with a file that isn't 0664 would be if one were created by an earlier version of smartnode. Even the old code passed 0664 to WriteFile. Do you remember more details?

Reply:
All I remember is that someone had an unusual `umask` config and it wouldn't actually write the file to `0664` until we explicitly chmod'd it with this line, and we figured no harm no foul. We can leave it out and see if anyone complains again, at least you'll have more info from a direct source then :P